### PR TITLE
Handle multiple senders for one event

### PR DESCRIPTION
**Pull request by MarcoLm993**
Implemented handling for corner case in which multiple FSMs send the same event. Update documentation of SCXML to Jani conversion

### CODE REVIEW

File: jani_generator/src/jani_generator/scxml_helpers/scxml_event.py [hunk 1]
```diff
@@ -55,26 +47,27 @@ def __init__(self,
                  data_struct: Optional[Dict[str, type]] = None):
         self.name = name
         self.data_struct = data_struct
-        self.senders: List[EventSender] = []
-        self.receivers: List[EventReceiver] = []
+        # Map automaton -> event name
+        # TODO: In the future, this could be a Dict[str, str] or even a Set, if we the action name
+        # is expected to match the event name.
```
Comment (ct2034) [inline jani_generator/src/jani_generator/scxml_helpers/scxml_event.py:52]: sorry, what does this mean?

File: jani_generator/src/jani_generator/scxml_helpers/scxml_event.py [hunk 2]
```diff
@@ -55,26 +47,27 @@ def __init__(self,
...
+        self.senders: Dict[str, EventSender] = {}
+        self.receivers: Dict[str, EventReceiver] = {}
```
Comment (ct2034) [inline jani_generator/src/jani_generator/scxml_helpers/scxml_event.py:54]: ♥

File: jani_generator/src/jani_generator/scxml_helpers/scxml_event_processor.py [hunk 3]
```diff
@@ -31,15 +31,18 @@
...
         timers: List[RosTimer],
-        jani_model: JaniModel):
+        jani_model: JaniModel) -> List[str]:
     """
     Implement the scxml events as jani syncs.
 
     :param events_holder: The holder of the events.
+    :param timers: The timers to add to the jani model.
     :param jani_model: The jani model to add the syncs to.
+    :return: The list of events having only senders
```
Comment (ct2034) [inline jani_generator/src/jani_generator/scxml_helpers/scxml_event_processor.py:41]:
```suggestion
    :return: The list of events having only senders.
```

File: jani_generator/src/jani_generator/scxml_helpers/scxml_event_processor.py [hunk 4]
```diff
@@ -52,49 +55,75 @@ def implement_scxml_events_as_jani_syncs(
...
-        event_automaton.add_edge(JaniEdge({
-            "location": "waiting",
-            "destinations": [{
+        event_action_names.append(event_name_on_send)
+        # Add the event handling automaton
+        jc.add_element(event_name)
+        if event.has_receivers():
+            # Add the receiver event as well
+            # TODO: Potential bug: if the same event is sent by multiple automata,
+            # the same action should be added multiple times in the composition
```
Comment (ct2034) [inline jani_generator/src/jani_generator/scxml_helpers/scxml_event_processor.py:69]:
```suggestion
```

File: jani_generator/test/test_systemtest_scxml_to_jani.py [hunk 5]
```diff
@@ -215,7 +215,7 @@ def _test_with_main(self, main_xml: str, folder: str, property_name: str, succes
         pos_res = "Result: 1" if success else "Result: 0"
         neg_res = "Result: 0" if success else "Result: 1"
         run_smc_storm_with_output(
-            f"--model {ouput_path} --property-name {property_name}",
+            f"--model {ouput_path} --properties-names {property_name}",
```
Comment (ct2034) [inline jani_generator/test/test_systemtest_scxml_to_jani.py:218]: Thats why you need the new smc-storm version, right?

Comment (ct2034) [reply jani_generator/test/test_systemtest_scxml_to_jani.py:218]: Double plural ?!

**Review by ct2034 — CHANGES_REQUESTED** [review]

File: jani_generator/test/test_systemtest_scxml_to_jani.py (hunk 5 above)
Comment (MarcoLm993) [reply jani_generator/test/test_systemtest_scxml_to_jani.py:218]: I wasn't sure, either... But the internet says it should be fine :)

File: jani_generator/src/jani_generator/scxml_helpers/scxml_event.py (hunk 1 above)
Comment (MarcoLm993) [reply jani_generator/src/jani_generator/scxml_helpers/scxml_event.py]: I made this clearer now! :)

**Review by ct2034 — APPROVED** [review]

**Pull request merged**